### PR TITLE
only perform isdigit check on non-negative values

### PR DESCRIPTION
**Pull request by levinli303**
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/isdigit-iswdigit-isdigit-l-iswdigit-l?view=msvc-170#return-value

required for MSVC in debug

### CODE REVIEW

File: src/celutil/greek.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ getFirstWordLength(std::string_view str)
         sp = str.length();
 
     // skip digits
-    while (sp > 0 && std::isdigit(str[sp - 1]) != 0)
+    while (sp > 0 && str[sp - 1] >= 0 && std::isdigit(str[sp - 1]) != 0)
```
Comment (375gnu) [inline src/celutil/greek.cpp:130]: `while (sp > 0 && std::isdigit(static_cast<unsigned char>(str[sp - 1])) != 0)`

**Review by 375gnu — CHANGES_REQUESTED** [review]

File: src/celutil/greek.cpp (hunk 1 above)
Comment (levinli303) [reply src/celutil/greek.cpp]: fixed

**Review by 375gnu — APPROVED** [review]

**Comment by ajtribick** [comment]
I believe the recommended fix here is to cast the argument to `unsigned char`, e.g. https://en.cppreference.com/w/cpp/string/byte/isdigit

> Like all other functions from [`<cctype`>](https://en.cppreference.com/w/cpp/header/cctype), the behavior of `std::isdigit` is undefined if the argument's value is neither representable as `unsigned char` nor equal to EOF. To use these functions safely with plain `char`s (or `signed char`s), the argument should first be converted to `unsigned char`:

>
> ```cpp
> bool my_isdigit(char ch)
> {
>     return std::isdigit(static_cast<unsigned char>(ch));
> }
> ```

**Comment by levinli303** [comment]
> I believe the recommended fix here is to cast the argument to `unsigned char`, e.g. https://en.cppreference.com/w/cpp/string/byte/isdigit

yeah I applied the fix, per 375gnu in the comment but did not update the PR title.

**Pull request merged**